### PR TITLE
Extended Support for API2600

### PR DESCRIPTION
**Pull request by AsisBagga**
### Description
Extended Support for API2600

### Issues Resolved
N/A

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass for go 1.11 + gofmt checks.
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: ov/server_hardware_type.go [hunk 1]
```diff
@@ -82,25 +117,26 @@ type StorageCapability struct {
...
+	Created             string               `json:"created,omitempty"`             // "created": "20150831T154835.250Z",
+	Description         utils.Nstring        `json:"description,omitempty"`         // "description": "ServerHardware",
+	ETAG                string               `json:"eTag,omitempty"`                // "eTag": "1441036118675/8",
+	Family              string               `json:"family,omitempty"`              // "family":"135",
+	FormFactor          string               `json:"formFactor,omitempty"`          // "formFactor":"HalfHeight",
+	Model               string               `json:"model,omitempty"`               // "model":"",
+	Modified            string               `json:"modified,omitempty"`            // "modified": "20150831T154835.250Z",
+	Name                string               `json:"name,omitempty"`                // "name": "ServerHardware 1",
+	PxeBootPolicies     []string             `json:"pxeBootPolicies,omitempty"`     // "pxeBootPolicies":{},
+	ServerDisplaySchema *serverDisplaySchema `json:"serverDisplaySchema,omitempty"` // "serverDisplaySchema":{},
```
Comment (nabhajit-ray) [inline ov/server_hardware_type.go:135]: ServerDisplaySchema is an array. Please check the api doc again

serverDisplaySchema A collection of display schemas of the server UI pages. 
array of BaseDisplaySettingsSchema
read only

Comment (nabhajit-ray) [reply ov/server_hardware_type.go:135]: Can you add the hypervisor cluster profile changes also. New fields got added to sharedstoragevolumes

Comment (AsisBagga) [reply ov/server_hardware_type.go]: done

File: ov/hypervisor_cluster_profile.go [hunk 2]
```diff
@@ -29,14 +29,37 @@ type HypervisorClusterProfile struct {
...
-	SharedStorageVolumes          []utils.Nstring                `json:"sharedStorageVolumes,omitempty"`          //"sharedStorageVolumes":"[]"
+	SharedStorageVolumes          []sharedStorageVolumes         `json:"sharedStorageVolumes,omitempty"`          //"sharedStorageVolumes":"[]"
 	State                         string                         `json:"state,omitempty"`                         //"state":"Active"
 	StateReason                   string                         `json:"stateReason,omitempty"`                   //"stateReason":"None"
 	Status                        string                         `json:"status,omitempty"`                        //"status":"OK"
 	Type                          string                         `json:"type,omitempty"`                          //"type":"HypervisorClusterProfileV4"
 	URI                           utils.Nstring                  `json:"uri,omitempty"`                           //"uri":"/rest/hypervisor-cluster-profiles/4340293c-0701-4773-b863-32854b0f7d29"
 }
 
+type sharedStorageVolumes struct {
```
Comment (nabhajit-ray) [inline ov/hypervisor_cluster_profile.go:40]: Any reason you have kept the struct name in smallcase. Smallcase wont allow it to be exported.

File: ov/hypervisor_cluster_profile.go [hunk 3]
```diff
@@ -29,14 +29,37 @@ type HypervisorClusterProfile struct {
...
+	ProtocolType            string                  `json:"protocolType, omitempty"`            //"protocolType":"",
+	ProvisionType           string                  `json:"provisionType, omitempty"`           //"provisionType":"",
+	RequestedCapacity       string                  `json:"requestedCapacity, omitempty"`       //"requestedCapacity":"",
+	StoragePoolUri          utils.Nstring           `json:"storagePoolUri, omitempty"`          //"storagePoolUri":"",
+	StorageVolumeUri        utils.Nstring           `json:"storageVolumeUri, omitempty"`        //"storageVolumeUri":"",
+	VolumeFileSystemType    string                  `json:"volumeFileSystemType, omitempty"`    //"volumeFileSystemType":"",
+	VolumeSource            string                  `json:"volumeSource, omitempty"`            //"volumeSource":"",
+}
+
+type hypervisorClusterVolume struct {
```
Comment (nabhajit-ray) [inline ov/hypervisor_cluster_profile.go:56]: Any reason you have kept the struct name in smallcase. Smallcase wont allow it to be exported.

File: ov/server_hardware_type.go [hunk 4]
```diff
@@ -70,6 +70,41 @@ type BiosSetting struct {
 	WarningText     string       `json:"warningText,omitempty"`     // "warningText":""
 }
 
+type serverDisplaySchema struct {
```
Comment (nabhajit-ray) [inline ov/server_hardware_type.go:73]: Any reason you have kept the struct name in smallcase. Smallcase wont allow it to be exported.

File: ov/server_hardware_type.go [hunk 5]
```diff
@@ -70,6 +70,41 @@ type BiosSetting struct {
...
+	Args        []args `json:"args,omitempty"`        // "args":{},
+	HelpText    string `json:"helpText,omitempty"`    // "helpText":"",
+	Label       string `json:"label,omitempty"`       // "label":"",
+	SettingType string `json:"settingType,omitempty"` // "settingType":"",
+}
+
+type args struct {
```
Comment (nabhajit-ray) [inline ov/server_hardware_type.go:80]: Same comment and below where you have used smallcase for struct names

File: ov/server_hardware_type.go (hunk 4 above)
Comment (AsisBagga) [reply ov/server_hardware_type.go]: done

File: ov/server_hardware_type.go (hunk 5 above)
Comment (AsisBagga) [reply ov/server_hardware_type.go]: done

File: ov/hypervisor_cluster_profile.go (hunk 3 above)
Comment (AsisBagga) [reply ov/hypervisor_cluster_profile.go]: done

File: ov/hypervisor_cluster_profile.go (hunk 2 above)
Comment (AsisBagga) [reply ov/hypervisor_cluster_profile.go]: updated

**Review by nabhajit-ray — APPROVED** [review]

**Review by VenkateshRavula — APPROVED** [review]

**Pull request merged**